app/client/models/ToggleEnterpriseModel.ts

```diff
@@ -48,37 +50,44 @@ export class ToggleEnterpriseModel extends Disposable {
     const task = async () => {
       await this._activationAPI.activateEnterprise(key);
       await retryOnNetworkError(() => this._configAPI.restartServer());
-      await this._reloadWhenReady();
     };
     await this._doWork(task);
   }
 
   private async _doWork(func: () => Promise<void>) {
     if (this.busy.get()) {
-      throw new Error("Please wait for the previous operation to complete.");
+      throw new Error(t("Please wait for the previous operation to complete."));
     }
+    this.busy.set(true);
     try {
-      this.busy.set(true);
       await this._notifier.slowNotification(func());
-    } finally {
+      await this._reloadWhenReady();
+    } catch (err) {
       this.busy.set(false);
+      throw err;
     }
   }
 
   private async _reloadWhenReady() {
-    // Now wait for the server to come back up, and refresh the page.
-    let maxTries = 10;
+    // Now wait about 30 seconds for the server to come back up, and
+    // refresh the page.
+    let maxTries = 30;
     while(maxTries-- > 0) {
       try {
-        await this._configAPI.getValue('edition');
+        const resp = await fetch(getHomeUrl() + "/status?ready=1");
```

Comment:
For client-side REST calls, using the `request` method of one of the `BaseAPI` descendants is preferable (it handles some tricky URL logic, and tracks pending requests for tests). Maybe we can add something to `InstallAPI` that calls this endpoint?

Reply:
Okay. I chose the configAPI instead, since `restart` was already there.